### PR TITLE
dev-lua/ldoc: New package ldoc

### PR DESCRIPTION
**Pull request by tomoyat1**
Package-Manager: portage-2.3.0
This is a new package for [LDoc](https://stevedonovan.github.io/ldoc/)
I am working on a bump for x11-wm/awesome-4.0, and it requires LDoc for building documentation. This PR attempts to satisfy that need.

Build log is [here](https://bpaste.net/show/7f101b564e14)

The patch is required because the makefile in LDoc assumes that /usr/bin (actually, the location of the lua binary) already exists, but because of how portage installs to a temp directory first, it doesn't. The patch adds a line to the makefile that creates the /usr/bin in the install directory.

Also I have set the package maintaner to be myself. Is this correct?

### CODE REVIEW

File: dev-lua/ldoc/ldoc-1.4.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.4.6-mkdir.patch"
+)
+
+S="${WORKDIR}/LDoc-${PV}"
+
+src_test() {
+	true
```
Comment (monsieurp) [inline dev-lua/ldoc/ldoc-1.4.6.ebuild]: If you wish to skip the `src_test()` phase, write `RESTRICT=test` below `S`.

File: dev-lua/ldoc/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>t@tomoyat1.com</email>
+		<name>Tomoya Tabuchi</name>
+	</maintainer>
```
Comment (monsieurp) [inline dev-lua/ldoc/metadata.xml]: Looks good. Make sure to add the proxy-maint email address to the metadata.xml file as well. Ping me if you need examples.

**Comment by monsieurp** [comment]
Hi and thanks for your contribution. I've made a couple of comments. Also, could you file a bug at https://bugs.gentoo.org to make sure your email address is in the bug tracker? Thanks!

File: dev-lua/ldoc/ldoc-1.4.6.ebuild (hunk 1 above)
Comment (tomoyat1) [inline dev-lua/ldoc/ldoc-1.4.6.ebuild]: Will fix.

**Comment by tomoyat1** [comment]
Hello,
I have made the two changes and have force pushed them. I have also created a [bug](https://bugs.gentoo.org/show_bug.cgi?id=603938) in the bug tracker.

File: dev-lua/ldoc/files/ldoc-1.4.6-mkdir.patch [hunk 3]
```diff
@@ -0,0 +1,10 @@
+--- LDoc-1.4.6/makefile	2016-11-07 23:20:55.000000000 +0900
++++ /tmp/makefile	2016-12-28 19:43:01.786472519 +0900
```
Comment (SoapGentoo) [inline dev-lua/ldoc/files/ldoc-1.4.6-mkdir.patch]: could you make your patch git-nice? i.e.:
```
--- a/makefile
+++ b/makefile
```
Less bytes, more readable

File: dev-lua/ldoc/ldoc-1.4.6.ebuild [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+DEPEND="dev-lua/penlight"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-1.4.6-mkdir.patch"
+)
+
+S="${WORKDIR}/LDoc-${PV}"
+RESTRICT=test
```
Comment (SoapGentoo) [inline dev-lua/ldoc/ldoc-1.4.6.ebuild]: please use `RESTRICT="!test? ( test )"` here, as `RESTRICT=test` is a Portage-ism and will for instance not work on Paludis

Comment (tomoyat1) [inline dev-lua/ldoc/ldoc-1.4.6.ebuild]: When I change that line, I get 
```
Error(s) in metadata for 'dev-lua/ldoc-1.4.6':
  RESTRICT: USE flag 'test' referenced in conditional '!test?' is not in IUSE
```

Comment (SoapGentoo) [inline dev-lua/ldoc/ldoc-1.4.6.ebuild]: Yes, just add `test` to `IUSE`

**Comment by tomoyat1** [comment]
@SoapGentoo I have made the changes you requested.

**Comment by SoapGentoo** [comment]
@tomoyat1 I bungled the `RESTRICT="test"` thing, hence I fixed it when merging, Thanks for the contribution.

**Pull request closed without merging**